pkg/controllers/backup/upload.go

```diff
@@ -24,7 +25,9 @@ func (h *handler) uploadToS3(backup *v1.Backup, objectStore *v1.S3ObjectStore, t
 		if err := os.MkdirAll(filepath.Join(tmpBackupGzipFilepath, objectStore.Folder), os.ModePerm); err != nil {
 			return removeTempUploadDir(tmpBackupGzipFilepath, err)
 		}
-		gzipFile = fmt.Sprintf("%s/%s", objectStore.Folder, gzipFile)
+		// we need to avoid both "//" inside the path and all leading and trailing "/"
+		gzipFile = fmt.Sprintf("%s/%s", strings.TrimRight(objectStore.Folder, "/"), gzipFile)
+		gzipFile = strings.Trim(gzipFile, "/")
```

Comment:
Is a TrimRight and a Trim necessary? Would it be better to TrimRight `//` and TrimLeft `/` for readability?

Reply:
Moreover, small nit, you could probably combine these unless you are worried about line length.

Reply:
the `TrimRight` in the following line is to remove the training `/` in `objectStore.Folder`, such that we will not have  gzipFile = `path//filename`
```
gzipFile = fmt.Sprintf("%s/%s", strings.TrimRight(objectStore.Folder, "/"), gzipFile)
```
the `Trim` in the following line is to remove all leading and trailing `/` and it is equivalent to `TrimLeft` + `TrimRight`
```
gzipFile = strings.Trim(gzipFile, "/")
```


@nickgerace  Please let me know if I answer your question.

Reply:
Forgive me, I understand they are logically equivalent. I was just curious if you preferred one way over the other for a reason. This looks good to me!